Update #27 - Refactored xts_allocator_client plugin

TB-1993 opened the pull request:
- Refactor the code to help it flow, it now has the following order:
  - Setup all the parser.
  - Parse the arguments.
  - Run the relevant command method, passing the parsed args.
- Upgraded and renamed the tests.
  - Renamed test_xts_allocator_client to test_xts_allocator_endpoints to better reflect what it is doing.
  - Turned test_copy into test_xts_client_positive_mocks since it is only testing positive paths using mocking

Closes #27

TB-1993 committed:
Update #27 - Refactored xts_allocator_client plugin

- Refactor the code to help it flow, it now has the following order:
  - Setup all the parser.
  - Parse the arguments.
  - Run the relevant command method passing the parsed args.
- Upgraded and renamed the tests.
  - Renamed test_xts_allocator_client to test_xts_allocator_endpoints to
    better reflect what it is doing.
  - Turned test_copy into test_xts_client_positive_mocks since it is
    only testing positive paths using mocking.

zghp commented on src/xts_core/plugins/xts_allocator_client.py:
```diff
@@ -196,277 +297,213 @@ def _get_rack_config(self, server: str, slot_id: str) -> dict:
+        if state:
+            payload['state'] = state
+        if owner_email:
+            payload = owner_email
```
payload['owner_email']

zghp commented on src/xts_core/plugins/xts_allocator_client.py:
```diff
@@ -67,76 +64,64 @@ def send_request(method, url, data=None):
+
+    @staticmethod
+    def _format_slots_list_to_table(response:list[dict]) -> Table:
+        table_headers = map(lambda x: x.capitalize(),response[0].keys())
```
what if there are no slots found? would it be worth to have a warning/ the table with a 'no slots found' message?

TB-1993 replied:
I have added a test to fix this now.

zghp commented on src/xts_core/plugins/xts_allocator_client.py:
```diff
@@ -31,32 +36,24 @@ class XTSAllocatorClient(BaseXTSPlugin):
+        '''
+        super().__init__()
 
     @property
```
shouldn't this be @ classmethod

TB-1993 replied:
I don't think this is necessary as it's a property. I'll be looking at changing this in my next ticket anyway

zghp commented on src/xts_core/plugins/xts_allocator_client.py:
```diff
@@ -31,32 +36,24 @@ class XTSAllocatorClient(BaseXTSPlugin):
+        '''Return a list of available CLI commands and their descriptions.'''
         return cls._positional_args
 
     @property
```
similar, @ classmethod

TB-1993 replied:
I don't think this is necessary as it's a property. I'll be looking at changing this in my next ticket anyway.

zghp commented on src/xts_core/plugins/xts_allocator_client.py:
```diff
@@ -150,44 +135,160 @@ def _allocate_slot(self, args: list):
-                sys.exit(1)
-            payload = {"platform": parsed_args.platform, "tags": parsed_args.tags}
+            if not platform:
+                plugin_utils.error('[default]--platform[/default] is required when [default]--id[/default] is not provided.')
```
should have sys.exit(1)

TB-1993 replied:
The sys.exit is built into the plugin_utils.error method.

TB-1993 committed:
Fix #27: Added docstring and fixed bug in list-slots

- Added missing docstrings in xts_allocator_client and tests
- Fixed bug in list-slots that occurred when the server had no slots
  configured.

TB-1993 committed:
Fix #27: Removed unreachable exit

TB-1993 committed:
Fix #27: Corrected _update_slot to use owner_email correctly.
